src/components/UniversalSearch/UniversalSearchProvider.tsx

```diff
@@ -0,0 +1,85 @@
+import {
+  IUniversalSearcher,
+  IUniversalSearcherResult,
+  UniversalSearchFilterMap,
+} from 'lib/UniversalSearcher/UniversalSearcher';
+import PropTypes from 'prop-types';
+import React, { ReactNode, useContext, useRef, useState } from 'react';
+import { useSelector } from 'react-redux';
+
+interface IUniversalSearchContextValue {
+  searchTerm: string;
+  search(newTerm: string): void;
+  search<T>(newTerm: string, type: string): void;
+  searchResults: IUniversalSearcherResult<unknown>[];
+  filters: UniversalSearchFilterMap;
+}
+
+const SEARCH_ITEMS_LIMIT = 5;
+const searchTermRegex = /^\S+$/;
+
+const initialContextValue: IUniversalSearchContextValue = {
+  searchTerm: '',
+  search: () => {},
+  searchResults: [],
+  filters: {},
+};
+
+const UniversalSearchContext = React.createContext<
+  IUniversalSearchContextValue
+>(initialContextValue);
+
+export function useUniversalSearch() {
```

Comment:
Exposing this hook, so we can use the search in any component

Reply:
Could be a fat arrow function since every other function in this file is also one.

```suggestion
export const useUniversalSearch () => {
```